Fix Omitted Bundle Download Location For Streaming Installations

Bastian-Krause opened the pull request:
Since the integration of RAUC streaming support in rauc-hawkbit-updater [1] the `bundle_download_location` config option can be ommited when the `stream_bundle` config option is enabled [2].

The value of the `bundle_download_location` config option is still used in streaming installation code paths leading to errors such as:
```
g_file_test: assertion 'filename != NULL' failed
```
To fix this, exit early in bundle clean up code and omit the check for free space if the `stream_bundle` config option is enabled.

We should omit the `bundle_download_location` config option for streaming installations in the install tests, because this is the critical code path for streaming installations. Specifying the option for this use case does not make a difference, so we do not need to test it.

[1] https://github.com/rauc/rauc-hawkbit-updater/pull/130
[2] fe30f971 ("hawkbit-client/config-file: implement http streaming support")

Fixes #149

ejoerns reviewed (approved):
Looks fine for me. :+1: 

And does not differ to much form what I had in preparation (https://github.com/ejoerns/rauc-hawkbit-updater/commit/d84ee2c2c8387330a734ef6e403a4d6a270763f5 ) :smirk: 
(Forgot to self-assign the issue obviously.)

Bastian-Krause commented:
> Looks fine for me.
> 
> And does not differ to much form what I had in preparation ([ejoerns@d84ee2c](https://github.com/ejoerns/rauc-hawkbit-updater/commit/d84ee2c2c8387330a734ef6e403a4d6a270763f5) ) (Forgot to self-assign the issue obviously.)

I like yours better, so I've cherry-picked and pushed it. I made the following changes:
- dropped the local variable `feedback_url` from `process_deployment()`
- tests: moved installation config to dedicated fixture
- tests: split the test adjustment into a dedicated commit

ejoerns reviewed (approved):
Like your remix. Better than both original versions! :+1:
